### PR TITLE
Changed isSoftware/forceSoftware to use the term 'fallback'

### LINKED ISSUE #2030: Rename forceSoftware and isSoftware
This issue is just about renaming these fields, not about changing their meaning or semantics.

There are two reasons:

1. There may be IP concerns about the WebGPU API if these names are left as-is.
2. These names describe a "how" but not a "what." They don't describe behavior or performance or anything measurable/observable.

We don't have any particular requests for what the name should be changed _to_, but it's fairly important to us that the name not be left including the word "software."

Ideas:
- `optimizedForHostTransfers`
- `optimizedForShaderBranching`

Our intention is not to muddle the meaning of these variables. It's totally fine to have normative prose for these values which says that the performance of these adapters is expected to match the performance of WebAssembly / JavaScript.

### PR DESCRIPTION
**Pull request by toji**
Fixes #2030.

This change replaces the flags `isSoftware` and `forceSoftware` with `isFallbackAdapter` and `forceFallbackAdapter` respectively, as well as changing up the language around the internal slots. "Fallback" was the term the editors felt most comfortable with after discussing the matter earlier today.

I find the change to be an unfortunate step backwards in terms of intuitiveness, but hopefully the definition offered as part of this change helps clear things up for developers while satisfying the desire expressed in #2030 to not use the terms "software" or "hardware". I do like that these terms allow the behavior and meaning of the flags to remain functionally identical.

***
<a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/pull/2039.html" title="Last updated on Aug 17, 2021, 1:51 AM UTC (75e7eb2)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2039/6831bfa...75e7eb2.html" title="Last updated on Aug 17, 2021, 1:51 AM UTC (75e7eb2)">Diff</a>

### CODE REVIEW

File: spec/index.bs
Comment (kainino0x) [inline spec/index.bs:1587]:
```suggestion
        When set to `true` indicates that only a [=fallback adapter=] may be returned. If the user
```

**Review by kainino0x — APPROVED** [review]
LGTM pending terminology approval from @litherum

**Review by litherum — APPROVED** [review]

**Comment by litherum** [comment]
Thank you!!

**Review by kvark — APPROVED** [review]

**Pull request merged**
